### PR TITLE
Module Phonebook

### PR DESCRIPTION
**Pull request by guillaumetavernier**
### Description

Please explain the changes you made here.

### Checklist

- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing
- [ ] Extended the documentation, if necessary

### CODE REVIEW

File: app/modules/phonebook/endpoints_phonebook.py [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+#                                     Logos                                    #
+# ---------------------------------------------------------------------------- #
+@module.router.post(
+    "/phonebook/associations/{association_id}/picture",
+    # response_model=standard_responses.Result,
+    status_code=201,
+)
+async def create_association_logo(
+    association_id: str,
+    image: UploadFile = File(...),
```
Comment (julien4215) [inline app/modules/phonebook/endpoints_phonebook.py:425]: This is not required anymore https://github.com/tiangolo/fastapi/releases/tag/0.78.0

**Review by julien4215 — CHANGES_REQUESTED** [review]
There is not init file for this module, we should add one.

File: app/modules/phonebook/endpoints_phonebook.py [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+@module.router.get(
+    "/phonebook/associations/{association_id}/picture",
+    response_class=FileResponse,
+    status_code=200,
+)
+async def read_association_logo(
+    association_id: str,
+    user: models_core.CoreUser = Depends(is_user_a_member),
+) -> FileResponse:
+    """
```
Comment (julien4215) [inline app/modules/phonebook/endpoints_phonebook.py:508]: We should check that the association_id is an association_id.

File: app/modules/phonebook/endpoints_phonebook.py [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+    Get the logo of an association.
+    """
+
+    return get_file_from_data(
+        directory="associations",
+        filename=str(association_id),
```
Comment (julien4215) [inline app/modules/phonebook/endpoints_phonebook.py:514]: Why are we casting this to a string. It is already a string.

**Review by julien4215 — CHANGES_REQUESTED** [review]

**Review by julien4215 — APPROVED** [review]

File: app/modules/phonebook/endpoints_phonebook.py [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+                memberships=member_memberships,
+                **member.__dict__,
+            ),
+        )
+    return members_complete
+
+
+@module.router.get(
+    "/phonebook/member/{user_id}",
+    response_model=schemas_phonebook.MemberComplete | None,
```
Comment (armanddidierjean) [inline app/modules/phonebook/endpoints_phonebook.py:267]: Prefer returning a 404

File: app/modules/phonebook/endpoints_phonebook.py [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+
+    await cruds_phonebook.delete_membership(membership_id, db)
+
+
+# ---------------------------------------------------------------------------- #
+#                                     Logos                                    #
+# ---------------------------------------------------------------------------- #
+@module.router.post(
+    "/phonebook/associations/{association_id}/picture",
+    # response_model=standard_responses.Result,
```
Comment (armanddidierjean) [inline app/modules/phonebook/endpoints_phonebook.py:453]: Commented line

**Review by armanddidierjean — APPROVED** [review]

File: app/modules/phonebook/cruds_phonebook.py [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+
+from sqlalchemy import delete, select, update
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from app.core import models_core
+from app.modules.phonebook import models_phonebook, phonebook_types, schemas_phonebook
+
+
+# ---------------------------------------------------------------------------- #
```
Comment (Petitoto) [inline app/modules/phonebook/cruds_phonebook.py:11]: We should be consistent on comments

**Review by Petitoto — CHANGES_REQUESTED** [review]

**Pull request merged**